src/dialects/builtin/op_interfaces.rs

```diff
@@ -185,3 +185,28 @@ pub trait OneResultInterface: Op {
         Ok(())
     }
 }
+
+/// An [Op] that calls a function
+pub trait CallOpInterface {
+    /// Returns the symbol of the callee of this call operation.
+    fn get_callee_sym(&self, ctx: &Context) -> String;
+}
+
+/// Returns the symbols of the callees of all the call operations in this operation.
+pub trait GetCalleesInterface<T: CallOpInterface + Op>: Op {
```

Comment:
Why does this have `T` ? Couldn't it just be `pub trait GetCalleesInterface: CallOpInterface {` ?

Reply:
On further thought, I realise that this isn't going to work. The interface is for an `Op`, but here you are digging in. 

I think this should just be a utility function (that takes an `Operation` as its argument) and not a trait at all.

Reply:
The concrete type for `T` is defined in impl for proc/func/etc op:
```rust
impl GetCalleesInterface<CallOp> for ProcOp {}
...
impl CallOpInterface for CallOp {
    fn get_callee_sym(&self, ctx: &Context) -> String {
        self.get_callee_sym(ctx)
    }
}
```

Reply:
hmmm, while this could work, why not just make `get_calles` a utility function that'll go deep and return symbols from any `Op` inside that implements the `CallOpInterface` ? That'll be simpler and more general right?

Reply:
Done.